tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -148,38 +160,10 @@ public void release(T element) {
    * @param element
    */
   public void discard(T element) {
```

Comment:
But shouldn't it be the "release" method to decide whether it can be reused or not instead of depending on the caller to decide? Seems not a good design to me.